rpm.go

```diff
@@ -426,34 +432,69 @@ func (r *RPM) writeGenIndexes(h *index) {
 	// rpm utilities look for the sourcerpm tag to deduce if this is not a source rpm (if it has a sourcerpm,
 	// it is NOT a source rpm).
 	h.Add(tagSourceRPM, EntryString(fmt.Sprintf("%s-%s.src.rpm", r.Name, r.FullVersion())))
-	if r.pretrans != "" {
-		h.Add(tagPretrans, EntryString(r.pretrans))
-		h.Add(tagPretransProg, EntryString("/bin/sh"))
+}
+
+func (r *RPM) writeSingleScriptlet(h *index, scriptlet, name string, scriptletTag, scriptletProgramTag int) error {
+	if scriptlet != "" {
+		scriptletProgram := r.scriptletProgram
+		// All this manual parsing instead of a simple regex to provide detailed error messages like rpmbuild:
```

Comment:
But.. We are not rpmbuild. One of our aims is simplicity.

I really appreciate the effort you put in here, so it's not easy for me to say so, but please remove all of this verification code.

rpmbuild is a very over-engineered and complex beast, riddled with backwards-compatibility-necessitated junk. It has a ton of implicit assumptions and checks. (And a lot of other problems too).

Our implementation is simple. We don't build anything. Originally we only supported rpms that are a "bag of files". Later we consented to adding scriptles. But verifying them goes a step too far.

If the user supplies a bad script, I guess it will fail for them?

Reply:
Okay, makes sense! What about going one step further and removing the `-p/path/here` syntax entirely? The rpmpack API could be extended to either have an additional `AddPostinInterpreter` (or `AddPostinScriptletProgram`?) for every scriptlet, just one or `SetScriptletProgramFor(what, path string)` where `what` is `postin`, `preun` etc. Either way, in `tar2rpm` just a single flag for all scriptlet types in the form of
 `-interpreter_for "preun:/bin/foo"` could be added.

Reply:
Yes!
The `-p` thing is messy.

I think `SetScriptletProgramFor` makes sense. I'm not sure what you propose by the single flag - how will a user specify separate interpreters?

Reply:
Done